### PR TITLE
fix: load cache error when CacheDecoder object is not callable

### PR DESCRIPTION
**Pull request by jamesgetx**
fix the code snippet `self._cache = json.load(f, cls=CacheDecoder(self.client))` raise `'CacheDecoder' object is not callable`

### CODE REVIEW

**Comment by yliaog** [comment]
thanks for the pr, could you please add a test case?

**Comment by jamesgetx** [comment]
> thanks for the pr, could you please add a test case?

Ok, I will add it later:)

**Comment by jamesgetx** [comment]
hi @yliaog. I have pushed the test case, but i'm not sure there is any other better solutions to test the code `self._cache = json.load(f, cls=partial(CacheDecoder, self.client))`

File: dynamic/test_discovery.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from . import DynamicClient
+
+
+class TestDiscoverer(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.config = base.get_e2e_configuration()
+
+    def test_init_cache_from_file(self):
```
Comment (yliaog) [inline dynamic/test_discovery.py:30]: so if you run the test without the fix, the test will fail, and there will be exception, right?

Comment (jamesgetx) [reply dynamic/test_discovery.py:30]: yes. without the fix, `_Discoverer__init_cache` will raise `'CacheDecoder' object is not callable` then call `self.invalidate_cache()`,  finally `self._write_cache()` will be called to modify `_Discoverer__cache_file`

**Pull request merged**